src/file/FileUtils.js

```diff
@@ -434,19 +434,24 @@ define(function (require, exports, module) {
      * @return {number} The result of the local compare function
      */
     function compareFilenames(filename1, filename2, extFirst) {
-        var ext1   = getFileExtension(filename1),
-            ext2   = getFileExtension(filename2),
-            lang   = brackets.getLocale(),
-            cmpExt = ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true}),
-            cmpNames;
+        var lang = brackets.getLocale();
         
-        if (brackets.platform === "win") {
-            filename1 = getFilenameWithoutExtension(filename1);
-            filename2 = getFilenameWithoutExtension(filename2);
+        function cmpExt() {
+            var ext1 = getFileExtension(filename1),
+                ext2 = getFileExtension(filename2);
+            return ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true});
+        }
+        
+        function cmpNames() {
+            if (brackets.platform === "win") {
```

Comment:
Sorry I didn't have a different term. But the only part I was saying to cache is the if statement where we are checking for the platform...  We don't need that check every time we run the code.

``` javascript
var _cmpNames = (function() {
   if (brackets.platform === 'window') {
     return  windows specific code;
   }
   else {
      return what we need;
   }
})();

function compareFilenames(filename1, filename2, extFirst) {
   function cmpNames() {
      return _cmpNames();
   }
}
```


Or some abstraction of this :)